src/Servers/Kestrel/Core/ref/Microsoft.AspNetCore.Server.Kestrel.Core.netcoreapp.cs

```diff
@@ -77,17 +77,24 @@ public partial class Http2Limits
         public int MaxRequestHeaderFieldSize { get { throw null; } set { } }
         public int MaxStreamsPerConnection { get { throw null; } set { } }
     }
+    public partial class Http3Limits
+    {
+        public Http3Limits() { }
+        public int HeaderTableSize { get { throw null; } set { } }
+        public int MaxRequestHeaderFieldSize { get { throw null; } set { } }
+    }
     [System.FlagsAttribute]
     public enum HttpProtocols
     {
         None = 0,
         Http1 = 1,
         Http2 = 2,
         Http1AndHttp2 = 3,
+        Http3 = 4,
     }
     public partial class KestrelServer : Microsoft.AspNetCore.Hosting.Server.IServer, System.IDisposable
     {
-        public KestrelServer(Microsoft.Extensions.Options.IOptions<Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions> options, Microsoft.AspNetCore.Connections.IConnectionListenerFactory transportFactory, Microsoft.Extensions.Logging.ILoggerFactory loggerFactory) { }
+        public KestrelServer(Microsoft.Extensions.Options.IOptions<Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions> options, System.Collections.Generic.IEnumerable<Microsoft.AspNetCore.Connections.IConnectionListenerFactory> transportFactories, Microsoft.Extensions.Logging.ILoggerFactory loggerFactory) { }
```

Comment:
@davidfowl I assume we don't this breaking change, want me to have two constructors?

Reply:
I'm not sure that would work with the DI containers, we might be stuck with a breaking change here...

Reply:
We could also have IMultiplexedConnectionFactory instead of IEnuermable, but yes it seems like we may need a breaking change.

Reply:
I talked about this a bit in #17037, but it might make better sense to inject any IConnectionListenerFactory and IMultiplexedConnectionFactory instances via separate constructor parameters. I think we should definitely do that if we end up making a IMultiplexedConnectionListener and MultiplexedConnectionContext so we don't end up repeatedly casting.

Reply:
See my reply. I think they should remain ConnectionContext.

Reply:
Let's push discussion to that issue and decide what to do there. I'm going to leave this as is and modify accordingly based on what we decide on the issue.

Reply:
Leave it as is